src/util/batch-process.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import Bottleneck from 'bottleneck';
+
+/** Concurrency presets based on some basic benchmarking & common-sense. */
+export const concurrencyPresets = {
+  npmPublish: 6,
+  fs: 10,
+  github: 16,
+};
+
+/**
+ * When running a batch process, return two collections: one of successful runs
+ * that completed without throwing an exception, and one where an exception
+ * occurred.
+ */
+export interface BatchProcessResponse<T> {
+  successes: Map<T, any>;
+  failures: Map<T, Error>;
+}
+
+/**
+ * Run some function of work over each item in an array (where the function is
+ * called with each item as an argument). Group the results by successes an
+ * failures based on whether an exception was thrown by that function.
+ */
+export async function batchProcess<T>(
```

Comment:
Nice wrapper for all those operations.  The options object also makes plugging in progress bar hooks later on that much easier too. +1